component/client/chat/ClientChat.tsx

```diff
@@ -48,15 +49,15 @@ class ClientChat extends React.Component<IClientChatProps, IClientChatState> {
       messages : []
     };
     room.setUrl(url, (data: IData, user) => {
-      this.setState((prevState, props) => {
+      this.setState((prevState: IClientChatState, props: IClientChatProps) => {
```

Comment:
Just note that that field might be redundant information. If the userID here will never get used, I might as well removed it.